data-prepper-api/src/main/java/com/amazon/dataprepper/model/configuration/PluginSetting.java

```diff
@@ -136,6 +137,88 @@ public String getStringOrDefault(final String attribute, final String defaultVal
+        }
+        throw new IllegalArgumentException(String.format(UNEXPECTED_ATTRIBUTE_TYPE_MSG, object.getClass(), attribute));
+    }
+
+    /**
+     * Returns the value of the specified Map<String, String> object, or {@code defaultValue} if this settings contains no value for
+     * the attribute.
+     *
+     * @param attribute    name of the attribute
+     * @param defaultValue default value for the setting
+     * @return the value of the specified attribute, or {@code defaultValue} if this settings contains no value for
+     * the attribute
+     */
+    public Map<String, String> getStringMapOrDefault(final String attribute, final Map<String, String> defaultValue) {
+        Object object = getAttributeOrDefault(attribute, defaultValue);
+        if (object == null) {
+            return null;
+        } else if (object instanceof Map) {
+            ((Map<?, ?>) object).forEach((key, value) -> {
+                if (!(key instanceof String) || !(value instanceof String)) {
+                    throw new IllegalArgumentException(String.format(UNEXPECTED_ATTRIBUTE_TYPE_MSG, object.getClass(), attribute));
+                }
+            });
+            return (Map<String, String>) object;
+        }
+        throw new IllegalArgumentException(String.format(UNEXPECTED_ATTRIBUTE_TYPE_MSG, object.getClass(), attribute));
+    }
+
+    /**
+     * Returns the value of the specified Map<String, List<String>>, or {@code defaultValue} if this settings contains no value for
+     * the attribute.
+     *
+     * @param attribute    name of the attribute
+     * @param defaultValue default value for the setting
+     * @return the value of the specified attribute, or {@code defaultValue} if this settings contains no value for
+     * the attribute
+     */
+    public Map<String, List<String>> getStringListMapOrDefault(final String attribute, final Map<String, List<String>> defaultValue) {
+        Object object = getAttributeOrDefault(attribute, defaultValue);
+        if (object == null) {
+            return null;
+        } else if (object instanceof Map) {
+            ((Map<?, ?>) object).forEach((key, value) -> {
+                if (key instanceof String) {
+                    if (value instanceof List){
+                        ((List<?>) value).forEach(val -> {
+                            if (!(val instanceof String)) {
+                                throw new IllegalArgumentException(String.format(UNEXPECTED_ATTRIBUTE_TYPE_MSG, object.getClass(), attribute));
+                            }
+                        });
+                    } else {
+                        throw new IllegalArgumentException(String.format(UNEXPECTED_ATTRIBUTE_TYPE_MSG, object.getClass(), attribute));
+                    }
+                } else {
+                    throw new IllegalArgumentException(String.format(UNEXPECTED_ATTRIBUTE_TYPE_MSG, object.getClass(), attribute));
+                }
+            });
```

Comment:
Instance check for String, List, Map is repeated in this class. Consider defining a function or java BiPredicate that takes in the object & type, and checks if they are of same type else throws exception. This code can be reused in different methods in this class.